ResearchKit/Common/ORKPasscodeStepViewController.m

```diff
@@ -647,4 +674,46 @@ - (BOOL)textFieldShouldEndEditing:(UITextField *)textField {
     return _shouldResignFirstResponder;
 }
 
+- (void)forgotPasscodeTapped
+{
+    if ([self.passcodeDelegate respondsToSelector:@selector(forgotPasscodeTapped:)])
+    {
+        [self.passcodeDelegate forgotPasscodeTapped:_forgotPasscodeButton];
+    }
+}
+
+- (BOOL) hasForgotPasscode
+{
+    if ([self.passcodeDelegate respondsToSelector:@selector(hasForgotPasscode)])
+    {
+        return [self.passcodeDelegate hasForgotPasscode];
+    }
+    return NO;
+}
+
+- (NSString*) forgotPasscodeButtonText
+{
+    if ([self.passcodeDelegate respondsToSelector:@selector(textForForgotPasscode)])
+    {
+        return [self.passcodeDelegate textForForgotPasscode];
+    }
+    return @"";
```

Comment:
Yup, totally get that and I've avoided language insertion for the same reason. In this case, you could probably use the `ORKLocalizedString(@"FORGOT_PASSWORD_BUTTON_TITLE", nil)` as the default. Passcode is a made-up word, anyway. :)